packages/algob/src/internal/deployer.ts

```diff
@@ -228,13 +241,31 @@ class DeployerBasicMode {
   }
 
   /**
-   * Returns cached program (from artifacts/cache) `ASCCache` object by filename.
-   * TODO: beta support - this will change
-   * @param name ASC name used during deployment
+   * Returns cached program (from artifacts/cache) `ASCCache` object by app/lsig name.
+   * @param name App/Lsig name used during deployment
    */
-  getDeployedASC (name: string): Promise<ASCCache | undefined> {
+  async getDeployedASC (name: string): Promise<ASCCache | AppCache | undefined> {
     const op = new CompileOp(this.algoOp.algodClient);
-    return op.readArtifact(name);
+
+    // app
+    const app = this.getAppByName(name);
+    if (app !== undefined) {
+      const approvalCache = await op.readArtifact(app.approvalFile);
```

Comment:
I not see `approvalFile` and `clearFile` declared in `rtypes.SSCInfo`.

Reply:
[https://github.com/scale-it/algo-builder/pull/570/files#diff-6a59e45f245e560b58d0eb568e95d8b96da9936493273edb36f12a0a292203d7R119](https://github.com/scale-it/algo-builder/pull/570/files#diff-6a59e45f245e560b58d0eb568e95d8b96da9936493273edb36f12a0a292203d7R119)

Reply:
I see my bad.